src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -236,6 +573,80 @@ Scalar<DataVector> spin_function(
+      get_surface_metric(spatial_metric, tangents, sin_theta);
+  const auto& inverse_surface_metric =
+      determinant_and_inverse(surface_metric).second;
+  const auto& trace_christoffel_second_kind = get_trace_christoffel_second_kind(
+      surface_metric, inverse_surface_metric, sin_theta, ylm);
+
+  const size_t matrix_dimension = get_matrix_dimension(ylm);
+  Matrix left_matrix(matrix_dimension, matrix_dimension, 0.0);
+  Matrix right_matrix(matrix_dimension, matrix_dimension, 0.0);
+  get_left_and_right_eigenproblem_matrices(
+      &left_matrix, &right_matrix, inverse_surface_metric,
+      trace_christoffel_second_kind, sin_theta, ricci_scalar, ylm);
+
+  DataVector eigenvalues_real_part(matrix_dimension, 0.0);
+  DataVector eigenvalues_im_part(matrix_dimension, 0.0);
+  Matrix eigenvectors(matrix_dimension, matrix_dimension, 0.0);
+  find_generalized_eigenvalues(&eigenvalues_real_part, &eigenvalues_im_part,
+                               &eigenvectors, left_matrix, right_matrix);
+
+  const std::array<DataVector, 3> smallest_eigenvectors =
+      get_eigenvectors_for_3_smallest_magnitude_eigenvalues(
+          eigenvalues_real_part, eigenvectors, ylm);
+
+  // Get normalized potentials (Kerr normalization) corresponding to the
+  // eigenvectors with three smallest-magnitude eigenvalues.
+  const auto& potentials =
+      get_normalized_spin_potentials(smallest_eigenvectors, ylm, area_element);
+
+  return get_spin_magnitude(potentials, spin_function, area_element, ylm);
+}
+
+template <typename Frame>
+std::array<double, 3> spin_vector(const double spin_magnitude,
+                                  const Scalar<DataVector>& area_element,
+                                  const Scalar<DataVector>& radius,
+                                  const tnsr::i<DataVector, 3, Frame>& r_hat,
+                                  const Scalar<DataVector>& ricci_scalar,
+                                  const Scalar<DataVector>& spin_function,
+                                  const YlmSpherepack& ylm) noexcept {
+  std::array<double, 3> spin_vector = {{0.0, 0.0, 0.0}};
+  auto integrand = make_with_value<Scalar<DataVector>>(get(radius), 0.0);
+  for (size_t i = 0; i < 3; ++i) {
+    // Compute horizon coordinates with a coordinate center such that
+    // the mass dipole moment vanishes.
+    get(integrand) =
+        get(area_element) * get(ricci_scalar) * r_hat.get(i) * get(radius);
+    get(integrand) =
+      ylm.definite_integral(get(integrand).data()) / (-8.0 * M_PI);
+    get(integrand) += r_hat.get(i) * get(radius);
+
+    // Get a component of a vector in the direction of the spin
+    get(integrand) *= get(area_element) * get(spin_function);
+    gsl::at(spin_vector, i) = ylm.definite_integral(get(integrand).data());
+  }
+
+  // Normalize spin_vector so its magnitude is the magnitude of the spin
+  double spin_vector_normalization = spin_magnitude / magnitude(spin_vector);
+  for (size_t i = 0; i < 3; ++i) {
```

Comment:
We have an `operator*` for std::array in `Utilities/StdArrayHelpers.hpp`. I think then this can become:

```
return spin_vector * (spin_magnitude / magnitude(spin_vector));
```